Fixes and cleanup

shawninder opened the pull request:
Our unsubscribe page uses this script, and the page was breaking because "p4_fb_page" was undefined, making the `attr` function throw an error which wasn't caught, interrupting script execution. This change inserts guards to avoid using `attr` with an undefined value. I took the occasion to remove all instances I found where a function could be called on an undefined variable. This includes places using `event.preventDefault()` or `event.stopPropagation()` without naming the first argument to the event listener function, effectively making this `undefined.preventDefault()` or `undefined.stopPropagation().

shawninder committed:
Fixes and cleanup

Our unsubscribe page uses this script, and the page was breaking because "p4_fb_page" was undefined, making the `attr` function throw an error which wasn't caught, interrupting script execution. This change inserts guards to avoid using `attr` with an undefined value. I also took the occasion to do a bit of clean up, reducing the amount of work javascript has to do as well as the number of lines. I also found places using `event.preventDefault()` or `event.stopPropagation()` without naming the first argument to the function, effectively making this `undefined.preventDefault()` or `undefined.stopPropagation(). This change also fixes that.

shawninder commented:
Using my personal github account here, but you can communicate with me at work using sinder@greenpeace.org or on Skype with "Shawn Inder"

comzeradd commented:
Hi @shawninder, thanks for the fix.

It's better if the commits are atomic and describe the change they introduce. So I'd prefer if you amend the commit message to something more descriptive and actually just include the fix (the `img` check).

The rest of the code changes can potentially happen in another PR. Although I'm not convinced yet. I understand that in some browsers this may be slightly faster, but also these changes make the code less readable.

shawninder commented:
Hi Nikos,

I actually find it more readable this way :p Just goes to show how
subjective readability can be sometimes, hehe. But you are in fact
absolutely right about atomic commits, my bad. I'll make another PR with
only the actual fix. I guess my programming best practices get a bit rusty
when I stop programming for months at a time :s

Shawn

On Mon, Dec 9, 2019 at 6:20 AM Nikos Roussos <notifications@github.com>
wrote:

> Hi @shawninder <https://github.com/shawninder>, thanks for the fix.
>
> It's better if the commits are atomic and describe the change they
> introduce. So I'd prefer if you amend the commit message to something more
> descriptive and actually just include the fix (the img check).
>
> The rest of the code changes can potentially happen in another PR.
> Although I'm not convinced yet. I understand that in some browsers this may
> be slightly faster, but also these changes make the code less readable.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/greenpeace/planet4-en-petitions/pull/6?email_source=notifications&email_token=AADWK6VZZEN4WI75YJBVQ73QXYSY7A5CNFSM4JU6FUG2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEGIZZXQ#issuecomment-563191006>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADWK6V4UM5HERXEGYUY2TTQXYSY7ANCNFSM4JU6FUGQ>
> .
>

shawninder committed:
Revert changes

I'm going to break this commit into multiple atomic commits which achieve a single thing at a time

shawninder committed:
Prevent the javascript from exiting prematurely

This script was at risk of exiting prematurely due to functions being called on potentially undefined variables.

shawninder commented:
@comzeradd The PR is now much more atomic, tackling the single problem of calling functions on potentially (or certainly) undefined variables. Is this OK?

comzeradd commented:
Thanks :+1: I merged it offline with 4a5452c9a0382643b11e18ed248c9f4e0581734e to squash the commits.
